lib/ProductOpener/Tags.pm

```diff
@@ -718,6 +719,35 @@ sub get_tags_grouped_by_property ($tagtype, $tagids_ref, $prop_name, $props_ref,
 	return $grouped_tags;
 }
 
+=head2 get_all_tags_having_property ($product_ref, $tagtype, $prop_name)
+
+For each tag of a given field ($tagtype, can be "labels" or "categories", for example),
+and a given property ($prop_name, without last column (:). Can be "incompatible_with:en", for example),
+return a hash of tagid <-> property_value
+remark: this DOES NOT handle property inheritance
+
+=head3 Return
+
+A hash, where keys are tagid and values are property_value
+
+=head4 Example, get_all_tags_having_property($product_ref, "labels", "incompatible_with:en")
+
+
+=cut
+
+sub get_all_tags_having_property ($product_ref, $tagtype, $prop_name) {
+	my %tag_property_hash = ();
+	foreach my $tagid (@{$product_ref->{$tagtype . "_tags"}}) {
```

Comment:
This will create en empty labels_tags / categories_tags array if it did not exist.
In Attributes.pm, if there's no labels_tags, then we consider we don't know if a product has the organic label or not. But if it has other labels (e.g. fair trade), then we consider that it if were organic, we would have the organic label too.
There's a bug in Attributes.pm that results in an empty labels_tags array being considered as having labels.